src/apme_engine/remediation/graph_engine.py

```
@@ -628,10 +628,31 @@ async def _propose_one(
                 return (node_id, fix, before_yaml)
 
-        # Fan out proposals concurrently
+        # Fan out proposals concurrently; emit per-node progress (ADR-068).
+        total_nodes = len(by_node)
+        progress_lock = asyncio.Lock()
+        completed_count = 0
+
+        async def _propose_one_tracked(
+            node_id: str,
+            node_violations: list[ViolationDict],
+        ) -> tuple[str, AINodeFix, str] | None | BaseException:
```

Ruff (0.16.0)

[warning] `None` not at the end of the type union.

Move `None` to the end of the type union

(RUF036)